scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/Configuration.kt

```diff
@@ -156,10 +163,11 @@ fun getKafkaProperties(params: KafkaStreamsParams): KafkaProperties {
         this[StreamsConfig.NUM_STREAM_THREADS_CONFIG] = 1
         this[StreamsConfig.SEND_BUFFER_CONFIG] = params.maxMessageSizeBytes
         this[StreamsConfig.RECEIVE_BUFFER_CONFIG] = params.maxMessageSizeBytes
+        // tell Kafka Streams to optimize the topology
+        this[StreamsConfig.TOPOLOGY_OPTIMIZATION_CONFIG] = StreamsConfig.OPTIMIZE
```

Comment:
I think setting OPTIMIZE to on is safe, and shouldn't significantly impact startup times. One of the areas where it can help is in reducing the need to rebalance consumers. At the moment the level of optimisations added is really small, and would not affect a wide array of cases. 

Only two cases are covered: 
1. reusing the source topic as the kafka-streams changelog topic (rather than creating a new one) and 
2. colapsing multiple repartition topics into a single repartition topic (this happens when changing the key that gets processed within a topology from one step to the next). 

Essentially some repartitions are lazy, waiting for a whole pass through the topology before doing one rebalance rather than eagerly doing rebalancing during each operation.